modules/library/main/src/main/java/org/geotools/feature/FeatureTypes.java

```diff
@@ -427,6 +433,34 @@ public static SimpleFeatureType newFeatureType(AttributeDescriptor[] types, Stri
         return ancestors;
     }
 
+    /**
+     * Whether the feature type has the specified name, or is a descendent from it
+     *
+     * @param featureType typeName with parentage in question
+     * @param name name to match against
+     * @return true if featureType has the same name, or is a descendent of the indicated name
+     */
+    public static boolean matches(FeatureType featureType, Name name) {
```

Comment:
Is this new functionality? Or was it hiding elsewhere...

Reply:
It was hidden in some other place, I pushed it up to a re-usable method.